src/planner/vectorfield/VectorFieldPlanner.cpp

```diff
@@ -196,29 +198,24 @@ std::unique_ptr<aikido::trajectory::Spline> planPathByVectorField(
            && terminationStatus
                   != VectorFieldPlannerStatus::CACHE_AND_TERMINATE);
 
-  if (cacheIndex >= 0)
+  // Print the termination condition.
+  if (terminationError)
   {
-    // Print the termination condition.
-    if (terminationError)
+    try
     {
-      try
-      {
-        std::rethrow_exception(terminationError);
-      }
-      catch (const VectorFieldTerminated& e)
-      {
-        dtwarn << "[VectorFieldPlanner::Plan] Terminated early: " << e.what()
-               << '\n';
-        return nullptr;
-      }
+      std::rethrow_exception(terminationError);
+    }
+    catch (const VectorFieldTerminated& e)
+    {
+      dtwarn << "[VectorFieldPlanner::Plan] Terminated early: " << e.what()
+             << '\n';
+      return nullptr;
     }
-
-    return convertToSpline(knots, cacheIndex, _stateSpace);
   }
-  // Re-throw whatever error caused planning to fail.
-  else if (terminationError)
+
+  if (cacheIndex >= 0)
   {
-    std::rethrow_exception(terminationError);
+    return convertToSpline(knots, cacheIndex, _stateSpace);
   }
   else
```

Comment:
Does this ever happen? We need to make sure to avoid throwing exceptions from these planning methods. I think we should be printing some warning and returning `nullptr` instead.

Reply:
It is possible but not currently reproducible in current test cases. 
It is a good idea to throw warning instead. I think all this part of code will be changed in a new version VFP.